ros_packages/pibsim_webots/pibsim_webots/pib_driver.py

```diff
@@ -1,148 +1,121 @@
-import rclpy
 import math
+from typing import Iterable, Tuple
+import rclpy
 from trajectory_msgs.msg import JointTrajectory, JointTrajectoryPoint
 
 
 class PibDriver:
+    """
+    This driver is the interface between pib's digital twin for Webots and the ROS jointTrajectory topic.
+    Every JointTrajectory message published to the topic gets converted into Webots movement instructions.
+    """
+
     def init(self, webots_node, properties):
+        """The init method is the Webot/ ROS node counterpart to the python __init__ constructor"""
         self.__robot = webots_node.robot
```

Comment:
Typically, private variables, methods, etc. in Python are marked by a single underscore instead of two. Two underscores are used for "name mangling", where Python will prefix the variable/method with the class name, e.g., here `__robot` becomes `_PibDriver__robot`.

This is typically done to prevent name clashes in inheritance, which does not apply here I think.

Applies to all instances in this file.

Reply:
I removed the double underscores as suggested